### ISSUE TITLE
Two factor authentication fails after 2.106.0

### ISSUE THREAD

**Issue by taisukeh** [fastlane/fastlane#13603]
### New Regression Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Regression Information

- Breaking version: `2.106.0`
- Last working version: `2.105.2`

### Regression Description

I can 2-fa login to App Store connect with the session cookie generated by fastlane spaceauth 2.105.2, but 
failed after 2.106.0 ( https://github.com/fastlane/fastlane/pull/13442 ). 

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>
$ bundle exec fastlane ios beta
Sending anonymous analytics information
Learn more at https://docs.fastlane.tools/#metrics
No personal or sensitive data is sent.
You can disable this by adding `opt_out_usage` at the top of your Fastfile

+-------------------+------------------------------------+
|            Summary for Setup Jenkins Action            |
+-------------------+------------------------------------+
| output_directory  | /sd/Slave/output       |
| derived_data_path | /sd/Slave/derived_data |
| result_bundle     | false                              |
+-------------------+------------------------------------+

Set output directory path to: "/sd/Slave/output".
Set derived data path to: "/sd/Slave/derived_data".
$ cp adep_profiles/profiles_uuid/*.mobileprovision ~/Library/MobileDevice/Provisioning\ Profiles
$ rm -rf adep_profiles
----------------------------------------
--- Step: Verifying fastlane version ---
----------------------------------------
Your fastlane version 2.107.0 matches the minimum requirement of 2.102.0  ✅
------------------------------
--- Step: default_platform ---
------------------------------
Driving the lane 'ios beta' 🚀
--------------------------
--- Step: xcode_select ---
--------------------------
Setting Xcode version to /Applications/Xcode/Xcode_10.app for all build steps
Init simulators ...
-----------------------
--- Step: cocoapods ---
-----------------------
$ bundle exec pod install
▸ Analyzing dependencies
▸ Downloading dependencies
▸ Generating Pods project
▸ Integrating client project
▸ Sending stats
▸ Pod installation complete! There are 0 dependencies from the Podfile and 0 total pods installed.
▸ [!] The Podfile does not contain any dependencies.
▸ [!] Automatically assigning platform `ios` with version `9.3` on target `FastlaneSample3` because no platform was specified. Please specify a platform for this target in your Podfile. See `https://guides.cocoapods.org/syntax/podfile.html#platform`.
▸ [!] Automatically assigning platform `ios` with version `9.3` on target `FastlaneSample3Tests` because no platform was specified. Please specify a platform for this target in your Podfile. See `https://guides.cocoapods.org/syntax/podfile.html#platform`.
▸ [!] Automatically assigning platform `ios` with version `9.3` on target `FastlaneSample3UITests` because no platform was specified. Please specify a platform for this target in your Podfile. See `https://guides.cocoapods.org/syntax/podfile.html#platform`.
▸ [!] Automatically assigning platform `watchos` with version `3.0` on target `Watch` because no platform was specified. Please specify a platform for this target in your Podfile. See `https://guides.cocoapods.org/syntax/podfile.html#platform`.
▸ [!] Automatically assigning platform `watchos` with version `3.0` on target `Watch Extension` because no platform was specified. Please specify a platform for this target in your Podfile. See `https://guides.cocoapods.org/syntax/podfile.html#platform`.
-----------------------------------
--- Step: dragon_carthage_cache ---
-----------------------------------
Generating .carthage_cache.yml file...
Checking Dragon for matching Carthage cache...
Downloading Carthage cache from Dragon...
$ cd /tmp && carthage_cache install '/Users/Shared/Jenkins/workspace/src/FastlaneSample3'
▸ Downloading archive with key '/sd/Slave/Library/Caches/carthage_cache/d397827ac5c12fbb2143fdfbe285e3a3235a6f8a9a8470071fd9c88df3c65fcd.zip'.
▸ Unarchiving '/sd/Slave/Library/Caches/carthage_cache/d397827ac5c12fbb2143fdfbe285e3a3235a6f8a9a8470071fd9c88df3c65fcd.zip' into '/Users/Shared/Jenkins/workspace/src/FastlaneSample3/Carthage/Build'.
--------------------------------
--- Step: get_version_number ---
--------------------------------
--------------------------------------------
--- Step: latest_testflight_build_number ---
--------------------------------------------
Login to App Store Connect (my-app-dev@gmail.com)
Two Factor Authentication for account 'my-app-dev@gmail.com' is enabled
Your session cookie has been expired.
Please enter the 6 digit code:
----------------------------
--- Step: upload_reports ---
----------------------------
+---------------------------+---------------------------------------------------------------+
|                                       mLane Context                                        |
+---------------------------+---------------------------------------------------------------+
| ORIGINAL_DEFAULT_KEYCHAIN | "/sd/Slave/Library/Keychains/ci-keychain-default" |
| DEFAULT_PLATFORM          | ios                                                           |
| PLATFORM_NAME             | ios                                                           |
| LANE_NAME                 | ios beta                                                      |
| VERSION_NUMBER            | 1.0                                                           |
+---------------------------+---------------------------------------------------------------+

+------+--------------------------------+-------------+
|                  fastlane summary                   |
+------+--------------------------------+-------------+
| Step | Action                         | Time (in s) |
+------+--------------------------------+-------------+
| 1    | import_from_git                | 0           |
| 2    | import_from_git                | 5           |
| 3    | Verifying fastlane version     | 0           |
| 4    | default_platform               | 0           |
| 5    | xcode_select                   | 0           |
| 6    | cocoapods                      | 4           |
| 7    | dragon_carthage_cache          | 2           |
| 8    | get_version_number             | 0           |
| 💥   | latest_testflight_build_number | 1           |
| 10   | upload_reports                 | 0           |
+------+--------------------------------+-------------+

fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

➡️  mFastlane match template_name
    https://github.com/fastlane/fastlane/issues/11881 [closed] 6 💬
    22 May 2018

➡️  mgym xcargs build-for-testing not working on scan xctestrun
    https://github.com/fastlane/fastlane/issues/10471 [closed] 3 💬
    24 Jan 2018

➡️  m`supported_platforms': [!] undefined method `split' for nil:NilClass (NoMethodError)
    https://github.com/fastlane/fastlane/issues/7141 [closed] 43 💬
    07 Mar 2017

🔗  You can ⌘ + double-click on links to open them directly in your browser.
mbundler: failed to load command: fastlane (/Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/bin/fastlane)
NoMethodError: m[!] undefined method `strip' for nil:NilClass
Did you mean?  String
Did you mean?  String
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/highline-1.7.10/lib/highline/question.rb:413:in `remove_whitespace'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/highline-1.7.10/lib/highline.rb:873:in `get_line'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/highline-1.7.10/lib/highline.rb:891:in `get_response'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/highline-1.7.10/lib/highline.rb:264:in `ask'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/spaceship/lib/spaceship/two_step_client.rb:67:in `handle_two_factor'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/spaceship/lib/spaceship/two_step_client.rb:43:in `handle_two_step'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/spaceship/lib/spaceship/client.rb:463:in `send_shared_login_request'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/spaceship/lib/spaceship/tunes/tunes_client.rb:124:in `send_login_request'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/spaceship/lib/spaceship/client.rb:667:in `do_login'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/spaceship/lib/spaceship/client.rb:363:in `login'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/spaceship/lib/spaceship/client.rb:71:in `login'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/spaceship/lib/spaceship/tunes/spaceship.rb:24:in `login'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/actions/app_store_build_number.rb:23:in `get_build_number'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/actions/app_store_build_number.rb:11:in `run'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/actions/latest_testflight_build_number.rb:11:in `run'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:232:in `block in execute_action'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/fast_file.rb:149:in `method_missing'
  Fastfile:61:in `block (2 levels) in parsing_binding'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/lane.rb:33:in `call'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/commander-4.4.6/lib/commander/command.rb:182:in `call'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/commander-4.4.6/lib/commander/command.rb:153:in `run'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/commander-4.4.6/lib/commander/runner.rb:446:in `run_active_command'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/commander-4.4.6/lib/commander/delegates.rb:15:in `run!'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/cli_tools_distributor.rb:108:in `take_off'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/gems/fastlane-2.107.0/bin/fastlane:23:in `<top (required)>'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/bin/fastlane:23:in `load'
  /Users/Shared/Jenkins/workspace/src/FastlaneSample3/.vendor/bundle/ruby/2.5.0/bin/fastlane:23:in `<top (required)>'

</pre>
</details>

### Environment

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                                                                       |
| --------------------------- | ------------------------------------------------------------------------------------------- |
| OS                          | 10.14                                                                                       |
| Ruby                        | 2.5.1                                                                                       |
| Bundler?                    | true                                                                                        |
| Git                         | git version 2.19.0                                                                          |
| Installation Source         | ~/.ghq/FastlaneSample3/vendor/bundle/ruby/2.5.0/bin/fastlane |
| Host                        | Mac OS X 10.14 (18A391)                                                                     |
| Ruby Lib Dir                | ~/.anyenv/envs/rbenv/versions/2.5.1/lib                                                     |
| OpenSSL Version             | OpenSSL 1.0.2p  14 Aug 2018                                                                 |
| Is contained                | false                                                                                       |
| Is homebrew                 | false                                                                                       |
| Is installed via Fabric.app | false                                                                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                                                 |
| Xcode Version               | 10.0                                                                                        |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             | 🚫 |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
fastlane_version "2.102.0"
default_platform :ios

platform :ios do
  scheme = "FastlaneSample3"
  workspace = "FastlaneSample3.xcworkspace"
  proj = "FastlaneSample3.xcodeproj"

  before_all do
    xcode_select ENV['XCODE_10_0'] if ENV['CI']

    cocoapods
  end

  desc "Submit a new Beta Build to Apple TestFlight"
  desc "This will also make sure the profile is up to date"
  lane :beta do
    target_version = get_version_number(xcodeproj: proj, target: "FastlaneSample3")
    latest_number = latest_testflight_build_number(version: target_version)
    increment_build_number(
      build_number: latest_number + 1
    )

    build_app(
      scheme: scheme,
      configuration: "Release",
      export_method: 'app-store'
    )

    pilot(skip_waiting_for_build_processing: true)
  end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
apple_id 'my-app-dev@gmail.com'
app_identifier "jp.co.my-app-dev.FastlaneSample3"
team_id = "XJP2NR4B5X"

[:beta, :release].each do |lane_name|
  for_lane lane_name do
    apple_id 'my-app-dev@gmail.com'
    app_identifier "jp.co.my-app-dev.FastlaneSample3.prod"
  end
end

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.107.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                            | Version      |
| ------------------------------ | ------------ |
| did_you_mean                   | 1.2.0        |
| bundler                        | 1.16.5       |
| CFPropertyList                 | 2.3.6        |
| concurrent-ruby                | 1.0.5        |
| i18n                           | 0.9.5        |
| minitest                       | 5.11.3       |
| thread_safe                    | 0.3.6        |
| tzinfo                         | 1.2.5        |
| activesupport                  | 4.2.10       |
| public_suffix                  | 2.0.5        |
| addressable                    | 2.5.2        |
| curb                           | 0.9.6        |
| fastimage                      | 2.1.4        |
| babosa                         | 1.0.2        |
| colored                        | 1.2          |
| highline                       | 1.7.10       |
| commander-fastlane             | 4.4.6        |
| dotenv                         | 2.5.0        |
| emoji_regex                    | 0.1.1        |
| excon                          | 0.62.0       |
| multipart-post                 | 2.0.0        |
| faraday                        | 0.15.3       |
| unf_ext                        | 0.0.7.5      |
| unf                            | 0.1.4        |
| domain_name                    | 0.5.20180417 |
| http-cookie                    | 1.0.3        |
| faraday-cookie_jar             | 0.0.6        |
| faraday_middleware             | 0.12.2       |
| gh_inspector                   | 1.1.3        |
| jwt                            | 2.1.0        |
| memoist                        | 0.16.0       |
| multi_json                     | 1.13.1       |
| os                             | 1.0.0        |
| signet                         | 0.11.0       |
| googleauth                     | 0.6.7        |
| httpclient                     | 2.8.3        |
| mime-types-data                | 3.2018.0812  |
| mime-types                     | 3.2.2        |
| declarative                    | 0.0.10       |
| declarative-option             | 0.1.0        |
| uber                           | 0.1.0        |
| representable                  | 3.0.4        |
| retriable                      | 3.1.2        |
| google-api-client              | 0.23.9       |
| json                           | 2.1.0        |
| mini_magick                    | 4.5.1        |
| multi_xml                      | 0.6.0        |
| plist                          | 3.4.0        |
| rubyzip                        | 1.2.2        |
| security                       | 0.1.3        |
| naturally                      | 2.2.0        |
| simctl                         | 1.6.5        |
| slack-notifier                 | 2.3.2        |
| terminal-notifier              | 1.8.0        |
| unicode-display_width          | 1.4.0        |
| terminal-table                 | 1.8.0        |
| tty-screen                     | 0.6.5        |
| tty-cursor                     | 0.6.0        |
| tty-spinner                    | 0.8.0        |
| word_wrap                      | 1.0.0        |
| atomos                         | 0.1.3        |
| claide                         | 1.0.2        |
| colored2                       | 3.1.2        |
| nanaimo                        | 0.2.6        |
| xcodeproj                      | 1.7.0        |
| rouge                          | 2.0.7        |
| xcpretty                       | 0.3.0        |
| xcpretty-travis-formatter      | 1.0.0        |
| badge                          | 0.9.1        |
| aws-sigv4                      | 1.0.3        |
| jmespath                       | 1.4.0        |
| aws-sdk-core                   | 2.11.157     |
| aws-sdk-resources              | 2.11.157     |
| aws-sdk                        | 2.11.157     |
| commander                      | 4.4.6        |
| carthage_cache                 | 0.9.1        |
| git_diff_parser                | 2.3.0        |
| thor                           | 0.20.0       |
| checkstyle_filter-git          | 1.1.0        |
| fastlane-plugin-badge          | 1.1.0        |
| fastlane-plugin-carthage_cache | 0.1.1        |
| trainer                        | 0.8.0        |
| fastlane-plugin-trainer        | 0.4.0        |
| saddler-reporter-text          | 1.0.0        |
| saddler                        | 1.0.0        |
| natural_number_string          | 1.0.1        |
| env_pull_request               | 1.1.0        |
| sawyer                         | 0.8.1        |
| octokit                        | 4.13.0       |
| nori                           | 2.6.0        |
| saddler-reporter-support       | 1.0.0        |
| env_branch                     | 1.3.0        |
| git                            | 1.5.0        |
| uri-ssh_git                    | 2.0.0        |
| git_clone_url                  | 2.0.0        |
| saddler-reporter-support-git   | 2.0.1        |
| saddler-reporter-github        | 2.0.0        |
| xcode-install                  | 2.4.4        |
| clamp                          | 0.6.5        |
| fuzzy_match                    | 2.0.4        |
| nap                            | 1.1.0        |
| cocoapods-core                 | 1.5.3        |
| cocoapods-deintegrate          | 1.0.2        |
| cocoapods-downloader           | 1.2.1        |
| cocoapods-plugins              | 1.0.0        |
| cocoapods-search               | 1.0.0        |
| cocoapods-stats                | 1.0.0        |
| netrc                          | 0.11.0       |
| cocoapods-trunk                | 1.3.1        |
| cocoapods-try                  | 1.1.0        |
| escape                         | 0.0.4        |
| fourflusher                    | 2.0.1        |
| molinillo                      | 0.6.6        |
| ruby-macho                     | 1.2.0        |
| cocoapods                      | 1.5.3        |
| mini_portile2                  | 2.3.0        |
| nokogiri                       | 1.8.4        |
| slather                        | 2.4.6        |
</details>

*generated on:* **2018-10-29**
</details>

**Comment by jamieaquila** [comment]
Any update on this?

**Comment by yuanjilee** [comment]
I run fastlane with terminal is OK. but with Jenkins error with "Two factor authentication fails"

**Comment by joshdholtz** [comment]
@taisukeh Heyyy 👋 I've been trying to replicate this issue for a while now I'm not having any luck 😕 Here is what I did to try to replicate:
1. Install _fastlane_ version `2.105.2`
2. `rm -rf ~/fastlane/spaceship/<my_email>`
    - To make sure I didn't have any other sessions from any other verions
3. Ran a lane that accessed my App Store Connect account (I used `lastest_testflight_build_number`)
    - Got a success
    - Got a cookie generated with `2.105.2` in `~/fastlane/spaceship/<my_email>`
4. Switched to latest _fastlane_ version `2.107.0`
5. Ran a lane that accessed my App Store Connect account (I used `lastest_testflight_build_number`)
    - Got a success with cookie generated in `2.105.2`

## Possible coincidence in timing with upgrade and expired cookie?
It looks like in your output that it shows your cookie has expired - `Your session cookie has been expired.` Have you tried running `rm -rf ~/fastlane/spaceship/<my_email>` to re-authenticating with 2FA to see if this works? 

I was not able to get that same `Your session cookie has been expired.` from my tests so maybe your cookie actually did expired? Do you know how long you have been using that cookie?

Sorry about not being able to help right away on this 😔 I will keep this tab open though to wait for your response to see if we can find either a solution to help you or a solution to help me replicate this 💪

**Comment by yuanjilee** [comment]
@joshdholtz  Thanks for your review. 
I face the similar problem as taisukeh.

I am troubled in **Two factor authentication**, I do as [spaceship](https://github.com/fastlane/fastlane/tree/master/spaceship). I set `FASTLANE_SESSION` to `EVN` with Jenkins , but it always tell me:
>Two Factor Authentication for account 'xxx@qq.com' is enabled
>Your session cookie has been expired.
>Please enter the 6 digit code: +------------------+-------------+

Here is my fastlane config:

```
 lane :release do
	  
        ENV["FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD"]= "lnjm-qvjy-wbsl-rscp"

 	 get_certificates           
	 get_provisioning_profile(app_identifier: "com.worktile.lesschat")
	 get_provisioning_profile(app_identifier: "com.worktile.lesschat.shareextension")

   scheme_name = "Worktile"
   configuration = 'Release'
   output_directory = File.expand_path("..", Dir.pwd) + File::Separator + 'build'

	 gym(
		clean: true,
	 	scheme: scheme_name,
                export_method: "app-store",
                configuration: configuration,
                output_directory: output_directory
	 )
   upload_to_testflight(skip_waiting_for_build_processing: true)
```

Here is my cookie:

```
---
- !ruby/object:HTTP::Cookie
  name: aa
  value: 18AAF1782FBF2672840791A095F17F81
  domain: idmsa.apple.com
  for_domain: true
  path: "/"
  secure: true
  httponly: true
  expires: 
  max_age: 
  created_at: &1 2018-11-05 11:31:35.595067000 +08:00
  accessed_at: *1
- !ruby/object:HTTP::Cookie
  name: DES5b4a2f8442ef42444908e42ce98f7db9e
  value: HSARMTKNSRVTWFlaNbjg2EdWXEWMBg83pz3DU+hBca4qBkVyQq1UNTNnQgvj+1fKAnzSkmItYi1bvi+eYpgG9URX+5Exx8ljSE/CgxXalN/0JK7xNdvFtpeQk03ymlhZ7A==SRVT
  domain: idmsa.apple.com
  for_domain: true
  path: "/"
  secure: true
  httponly: true
  expires: 
  max_age: 2592000
  created_at: 2018-10-26 15:12:29.409003000 +08:00
  accessed_at: 2018-11-05 11:31:33.497082000 +08:00
- !ruby/object:HTTP::Cookie
  name: dslang
  value: US-EN
  domain: apple.com
  for_domain: true
  path: "/"
  secure: true
  httponly: true
  expires: 
  max_age: 
  created_at: 2018-11-05 11:31:35.595228000 +08:00
  accessed_at: &2 2018-11-05 11:31:35.596738000 +08:00
- !ruby/object:HTTP::Cookie
  name: site
  value: USA
  domain: apple.com
  for_domain: true
  path: "/"
  secure: true
  httponly: true
  expires: 
  max_age: 
  created_at: 2018-11-05 11:31:35.595443000 +08:00
  accessed_at: *2
- !ruby/object:HTTP::Cookie
  name: acn01
  value: fda3WLFen1KEvQflA2opXceQ8X633G6aABs3sc1afEo=
  domain: apple.com
  for_domain: true
  path: "/"
  secure: true
  httponly: true
  expires: 
  max_age: 31536000
  created_at: 2018-11-05 11:31:35.595784000 +08:00
  accessed_at: *2
- !ruby/object:HTTP::Cookie
  name: myacinfo
  value: DAWTKNV2d491610eba13941c051d772f9a3f9687cc35706e397d885f26e70633b1db1f391c68b18e37fe26d8036a4230637670d6445678482531dc13bf2b7c444d453348c74334f55bffb2d17d071891fda4794820d3fd6ac2f4291cb8f5380865dca3491661d54f0380a126248de576108ab4147f9b6be7f2cd5a7d6df37ef18d4689dbef39b54a0fa998fb1e4e2e9dffb4cd3e30ef77529fc8283aedc1460e70012eb97ff1202f18880a5dc144a5429f3cd0b39d79406c1c24df03ce813dca3a8db2924360b3a9bceeb84f40e88750aafe643058d9bb8aa5a77108c55f21563b152b77bd051ff6375ba6b406e876d58312870363643930316238393261386164633963633562653830323635393035326463333039663465306464MVRYV2
  domain: apple.com
  for_domain: true
  path: "/"
  secure: true
  httponly: true
  expires: 
  max_age: 
  created_at: 2018-11-05 11:31:35.595991000 +08:00
  accessed_at: *2
- !ruby/object:HTTP::Cookie
  name: itctx
  value: eyJjcCI6MjE5MDcyOSwiZHMiOjgwNDM4NDc3NTQsImV4IjoiMjAxOC0xMS01IDExOjMxOjM2In0|ns7ulm66e4568u9a1ji7vrqns0|54uc0eh_h3AAbx7nK6NLB55s8Bk
  domain: apple.com
  for_domain: true
  path: "/"
  secure: true
  httponly: true
  expires: 
  max_age: 
  created_at: &3 2018-11-05 11:31:36.577430000 +08:00
  accessed_at: *3
- !ruby/object:HTTP::Cookie
  name: itcdq
  value: '0'
  domain: apple.com
  for_domain: true
  path: "/"
  secure: true
  httponly: false
  expires: 
  max_age: 
  created_at: 2018-10-31 12:25:51.303721000 +08:00
  accessed_at: *2
- !ruby/object:HTTP::Cookie
  name: dc
  value: "-1"
  domain: itunes.apple.com
  for_domain: true
  path: "/"
  secure: false
  httponly: false
  expires: 
  max_age: 3600
  created_at: &4 2018-11-05 11:31:36.577212000 +08:00
  accessed_at: *4
- !ruby/object:HTTP::Cookie
  name: dqsid
  value: eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJqdGkiOiJ0QVVHQVpHZnJmX09CTlN2WmVGdzhnIn0.YCp4t09azYSBMoLD05b2RD3mKEpTyWUwA4IH6PNlqds
  domain: olympus.itunes.apple.com
  for_domain: false
  path: "/"
  secure: true
  httponly: true
  expires: 
  max_age: 1800
  created_at: &5 2018-11-05 11:31:36.577025000 +08:00
  accessed_at: *5

```

I don't know where I should to set. another? so sad. 
THANKS.

**Comment by taisukeh** [comment]
I could reproduce this problem  in the following steps.

1. Install *fastlane* version `2.107.0`
1. `rm -rf ~/.fastlane/spaceship/`
1. `bundle exec fastlane spaceauth -u <my-email>`
1. `export export FASTLANE_SESSION=...`
1. `rm -rf ~/.fastlane/spaceship/`
    - Remove cache generated by spaceauth
1. Ran a lane that accessed my App Store Connect account (I used lastest_testflight_build_number)
    - Got a failure

I suspect  this problem is `FASTLANE_SESSION` value affected by https://github.com/fastlane/fastlane/pull/13442. 

With `2.105.2`, got a success.

1. Install *fastlane* version `2.105.2`
1. `rm -rf ~/.fastlane/spaceship/`
1. `bundle exec fastlane spaceauth -u <my-email>`
1. `export export FASTLANE_SESSION=...`
1. `rm -rf ~/.fastlane/spaceship/`
    - Remove cache generated by spaceauth
1. Ran a lane that accessed my App Store Connect account (I used lastest_testflight_build_number)
    - Got a success

Thanks for helping out in this issue 🙇

**Comment by dlerman2** [comment]
Same problem here. Confirmed that login with FASTLANE_SESSION worked on 2.105.2 but fails on 2.107.0, following the steps outlined by @taisukeh.

**Comment by yuanjilee** [comment]
@jarrodparkes  Thanks for your response.  

I do test as you reply follow @taisukeh  :


1. Install fastlane version 2.108.0
2. rm -rf ~/.fastlane/spaceship/
3. bundle exec fastlane spaceauth -u <my-email>
4. export FASTLANE_SESSION=...
5. rm -rf ~/.fastlane/spaceship/
Remove cache generated by spaceauth
6. Ran a lane that accessed my App Store Connect account in CI (I used lastest_testflight_build_number)
Got a failure

These EVN I should set in CI `FASTLANE_PASSWORD`, `FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD `, `FASTLANE_SESSION `, but also failed with print 

```
Starting login with user 'xxx@email.com'
Two Factor Authentication for account 'xxx@ email.com' is enabled
Your session cookie has been expired.
Please enter the 6 digit code: +------------------+-------------+
```

And there is a 6 digit code in my device, but I can't input with CI.

All is good in local, but failed with CI(Jenkins).

**Comment by jarrodparkes** [comment]
@yuanjilee when you generated the `FASTLANE_SESSION`, what version of fastlane was running? based on your list of items, you would have been using v2.108.0. you should uninstall that version, and run v2.105.2 like @taisukeh recommended.

TL;DR - make sure you generate the `FASTLANE_SESSION` with fastlane v2.105.2

**Comment by kevinsperrine** [comment]
I rolled back Fastlane and regenerated the cookie with 2.105.2, but am now running into this error when running match. I've triple checked the FASTLANE_SESSION variable on BitRise. Anyone else still have problems? ```
INFO [2018-11-12 09:03:52.07]: Verifying that the certificate and profile are still valid on the Dev Portal...
Loading session from environment variable
Error loading session from environment
Make sure to pass the session in a valid format
```

**Comment by kevinsperrine** [comment]
Fixed by taking the entire output of `~/.fastlane/spaceship/[REDACTED]/cookie`, replacing the newlines with `\n` and shoving it all into the FASTLANE_SESSION variable on bitrise.
